operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/AbstractControllerConfiguration.java

```diff
@@ -1,10 +1,11 @@
 package io.javaoperatorsdk.operator.api.config;
 
-import io.fabric8.kubernetes.client.CustomResource;
 import java.util.Collections;
 import java.util.Set;
 
-public abstract class AbstractControllerConfiguration<R extends CustomResource>
+import io.fabric8.kubernetes.client.CustomResource;
+
+public abstract class AbstractControllerConfiguration<R extends CustomResource<?, ?>>
```

Comment:
wonder if this need to be an abstract class or just have some more params to set the crd class name and the configuration service

Reply:
Maybe… the reason why it's abstract is that it was needed like that at some point for the Quarkus extension. Maybe it doesn't need to be abstract anymore. Let me check.

Reply:
It appears it's not needed anymore. I have local changes to this PR to account for changes made for #499 that I intend to merge first. I will make the changes to this PR after the other one is merged.

Reply:
🤔 Cleaning this up properly, though, would require a breaking API change because, obviously, the class shouldn't be still named `AbstractControllerConfiguration` if it's not abstract anymore.

Reply:
I fixed the backwards compatibility issue but name has to stay as-is for now.